torch/nn/functional.py

```diff
@@ -1408,6 +1408,22 @@ def linear(input, weight, bias=None):
     return ret
 
 
+@weak_script
+def linear_weight_float(input, weight, bias=None):
```

Comment:
if this is only for MultiHeadAttention, prefix with `_` and make it internal. Right now, the API and semantics of this function dont look great.

Reply:
The function is not needed any more.